### DISCUSSION TITLE
Run subprocess and recieve result

### DISCUSSION THREAD

**Discussion by abkein** [fastapi/fastapi#6859] [Questions] [answered]
### First check

* [+] I added a very descriptive title to this issue.
* [+] I used the GitHub search to find a similar issue and didn't find it.
* [+] I searched the FastAPI documentation, with the integrated search.
* [+] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [+] I already read and followed all the tutorial in the docs and didn't find an answer.
* [+] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [+] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [+] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [+] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Description

I want to implement something like this:

```Python
def runandget():
    myprocess = subprocess.run(["node", "./js/duck.js"], stdout=subprocess.PIPE)
    thoutput = myprocess.stdout.decode('utf-8')
    return thoutput

@app.post("/doit")
def dothis():
    resultfunc = runandget()
    return {"result": resultfunc}
```
But `runandget()` finishes instantly and variable `resultfunc`  becomes empty

If I run function `runandget()` separetely (for example, from terminal), then everything works correctly

### Environment

* OS: [Debian GNU/Linux 10 (buster)]:
* FastAPI Version [0.63.0]:
* Python version [3.7.3]:

**Comment by Kludex** [comment, marked as answer]
It works just fine... :kissing: 

The example I was able to run:
```python
import subprocess

from fastapi import FastAPI

app = FastAPI()

def runandget():
    myprocess = subprocess.run(["ls"], stdout=subprocess.PIPE)
    thoutput = myprocess.stdout.decode('utf-8')
    return thoutput

@app.post("/doit")
def dothis():
    resultfunc = runandget()
    return {"result": resultfunc}
```

**Comment by abkein** [comment]
I'm so sorry, there is my error in javascript. Thank you for responding

**Comment by rudSarkar** [comment]
Thanks @Kludex it helped :heart:
